gazebo/rendering/CustomPSSMShadowCameraSetup.cc

```diff
@@ -624,6 +649,10 @@ void CustomPSSMShadowCameraSetup::getShadowCamera(const Ogre::SceneManager *_sm,
   // restore near/far
   cam->setNearClipDistance(oldNear);
   cam->setFarClipDistance(oldFar);
+  if (cam->isCustomProjectionMatrixEnabled())
+  {
+    cam->setCustomProjectionMatrix(true, oldCustomProjMat);
```

Comment:
This looks like we're resetting the custom projection matrix back to the old matrix. So we only want to modify the custom projection matrix temporarily (before calling the clip distance methods)?

Reply:
Correct. In the case of non-custom projection matrices, lines 625 and 626 build a new temporary matrix and lines 650 and 651 rebuild the original matrix. Those lines have no effect on a custom matrix, so I found a different way to build a temporary matrix and restore the original.

Line 619 points out that this is all very "hacky," so the original author of this code probably would have preferred a cleaner solution but didn't want all the refactoring that it would have required.